full-service/src/service/transaction.rs

```diff
@@ -147,6 +155,22 @@ impl From<mc_ledger_db::Error> for TransactionServiceError {
     }
 }
 
+impl From<hex::FromHexError> for TransactionServiceError {
+    fn from(src: hex::FromHexError) -> Self {
+        Self::FromHexError(src)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub enum TransactionMemo {
+    /// Recoverable Transaction History memo.
+    RTH,
+
+    /// Burn Redemption memo, with an optional 64 byte redemption memo hex
+    /// string.
+    BurnRedemption(Option<String>),
```

Comment:
Why is the memo optional?
Separately, I think it should be `BurnRedemption([u8; BurnRedemptionMemo::MEMO_DATA_LEN])`, we should avoid passing invalid/unparseable memos as early as possible.